src/dune/file_tree.ml

```diff
@@ -242,7 +248,13 @@ end = struct
 
   let t = Fdecl.create to_dyn
 
-  let set = Fdecl.set t
+  let set x =
+    match Fdecl.peek t with
+    | None -> Fdecl.set t x
+    | Some x' ->
+      if not (equal x x') then
+        (* The next call will fail, but will give a food error message *)
```

Comment:
"good error message"? :-)

Reply:
Well, it's actually not possible for the user to change these variables once they run dune. So it would just be a programming bug and this message should suffice for us to fix the problem.

Reply:
Fair enough.